File: practice/convert/example2_book_library/book-library/src/main/java/dev/starryeye/book_library/application/book/command/RegisterBookService.java
package dev.starryeye.book_library.application.book.command;

import dev.starryeye.book_library.application.book.command.command.RegisterBookCommand;
import dev.starryeye.book_library.domain.book.Book;
import dev.starryeye.book_library.domain.book.BookRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Transactional
@Service
@RequiredArgsConstructor
public class RegisterBookService {

    private final BookRepository bookRepository;

    public void register(RegisterBookCommand command) {

        Book book = Book.createBook(
                command.bookname()
        );

        bookRepository.save(book);
    }
}

File: practice/convert/example2_book_library/kotlin/book-library/src/main/kotlin/dev/starryeye/book_library/application/book/command/RegisterBookService.kt
package dev.starryeye.book_library.application.book.command

import dev.starryeye.book_library.domain.book.BookRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Transactional
@Service
class RegisterBookService(
    private val bookRepository: BookRepository,
) {
}